airbyte-integrations/connectors/source-mongodb-internal-poc/src/test-integration/java/io/airbyte/integrations/source/mongodb/internal/MongoDbSourceAcceptanceTest.java

```diff
@@ -162,6 +204,165 @@ public void testIncrementalReadSelectedColumns() throws Exception {
     verifyFieldNotExist(records, COLLECTION_NAME, TEST_ARRAY_FIELD);
   }
 
+  @Test
+  void testSyncEmptyCollection() throws Exception {
+    final ConfiguredAirbyteCatalog configuredCatalog = getConfiguredCatalog();
+    final AirbyteStream otherAirbyteStream = MongoCatalogHelper.buildAirbyteStream(OTHER_COLLECTION_NAME, DATABASE_NAME,
+        List.of(Field.of(NAME_FIELD, JsonSchemaType.STRING), Field.of(INT_TEST_FIELD, JsonSchemaType.NUMBER)));
+    configuredCatalog.withStreams(List.of(convertToConfiguredAirbyteStream(otherAirbyteStream, SyncMode.INCREMENTAL)));
+
+    final List<AirbyteMessage> messages = runRead(configuredCatalog);
+    final List<AirbyteRecordMessage> recordMessages = filterRecords(messages);
+    final List<AirbyteStateMessage> stateMessages = filterStateMessages(messages);
+
+    assertEquals(0, recordMessages.size());
+    assertEquals(1, stateMessages.size());
+
+    final AirbyteStateMessage lastStateMessage = Iterables.getLast(stateMessages);
+    assertNotNull(lastStateMessage.getGlobal().getSharedState());
+    assertFalse(lastStateMessage.getGlobal().getSharedState().isEmpty());
+    assertTrue(lastStateMessage.getGlobal().getStreamStates().isEmpty());
+  }
+
+  @Test
+  void testCDCStreamCheckpointingWithMultipleStreams() throws Exception {
+    final int otherCollectionCount = 100;
+    insertData(DATABASE_NAME, OTHER_COLLECTION_NAME, otherCollectionCount);
+
+    final ConfiguredAirbyteCatalog configuredCatalog = getConfiguredCatalog();
+    final List<ConfiguredAirbyteStream> streams = configuredCatalog.getStreams();
+    final AirbyteStream otherAirbyteStream = MongoCatalogHelper.buildAirbyteStream(OTHER_COLLECTION_NAME, DATABASE_NAME,
+        List.of(Field.of(NAME_FIELD, JsonSchemaType.STRING), Field.of(INT_TEST_FIELD, JsonSchemaType.NUMBER)));
+    streams.add(convertToConfiguredAirbyteStream(otherAirbyteStream, SyncMode.INCREMENTAL));
+    configuredCatalog.withStreams(streams);
+
+    // Start a sync with two streams
+    final List<AirbyteMessage> messages = runRead(configuredCatalog);
+    final List<AirbyteRecordMessage> recordMessages = filterRecords(messages);
```

Comment:
Can we assert the contents of the record message as well. Also assert that the record messages contain the metadata (like the cursor field value) that should be added.